certification/internal/shell/has_license.go

```diff
@@ -1,35 +1,48 @@
 package shell
 
 import (
-	"os/exec"
 	"strings"
 
 	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/cli"
 	log "github.com/sirupsen/logrus"
 )
 
 type HasLicenseCheck struct{}
 
 func (p *HasLicenseCheck) Validate(image string) (bool, error) {
-	stdouterr, err := exec.Command("podman", "run", "-it", "--rm", "--entrypoint", "ls", image, "-A", "/licenses").CombinedOutput()
-	result := string(stdouterr)
+	line, err := p.getDataToValidate(image)
 	if err != nil {
-		if strings.Contains(result, "No such file or directory") || result == "" {
-			log.Warn("license not found in the container image at /licenses")
-			return false, nil
-		}
-
-		log.Error("some error attempting to identify if /licenses container the license: ", err)
 		return false, err
 	}
+	return p.validate(line)
+}
 
-	// sanity check - in case we don't get an error, but also don't have the file.
-	if strings.Contains(result, "No such file or directory") || result == "" {
-		log.Warn("license not found in the container image at /licenses")
-		return false, nil
+func (p *HasLicenseCheck) getDataToValidate(image string) (string, error) {
+	runOpts := cli.ImageRunOptions{
+		EntryPoint:     "ls",
+		EntryPointArgs: []string{"-A", "/licenses"},
+		LogLevel:       "debug",
+		Image:          image,
 	}
+	runReport, err := podmanEngine.Run(runOpts)
+	if err != nil {
+		ok, _ := p.validate(runReport.Stdout)
+		if ok {
+			log.Error("some error attempting to identify if /licenses container the license: ", err)
+			log.Debugf("Stdout: %s", runReport.Stdout)
+			log.Debugf("Stderr: %s", runReport.Stderr)
+			return "", err
+		}
+
+	}
+	line := runReport.Stdout
+	return line, nil
```

Comment:
you can combine these two lines; no need to introduce new var if it is not being reused.